addons/mail/static/src/models/suggested_partners/suggested_partners.js

```diff
@@ -0,0 +1,53 @@
+odoo.define('mail/static/src/models/suggested_partners/suggested_partners.js', function (require) {
+'use strict';
+
+const { registerNewModel } = require('mail/static/src/model/model_core.js');
+const { attr, many2one } = require('mail/static/src/model/model_field.js');
+const mailUtils = require('mail.utils');
+
+function factory(dependencies) {
+    class SuggestedPartner extends dependencies['mail.model'] {
+        init(...args) {
+            super.init(...args);
+        }
+
+        static convertData(data) {
+            const parsedEmail = data[1] && mailUtils.parseEmail(data[1]);
+            const data2 = {
+                checked: true,
+                partner_id: data[0] ? data[0] : undefined,
+                reason: data[2],
+            };
+
+            if (data2.partner_id) {
+                data2.partner = [
+                    ['insert', {
+                        id: data2.partner_id,
+                        name: parsedEmail[0],
+                        email: parsedEmail[1],
+                        email_formatted: data[1],
+                    }],
+                ];
+            } else {
+                data2.name = '"' + parsedEmail[0] + '" (' + parsedEmail[1] + ')';
+                data2.email = parsedEmail[1];
+            }
+
+            return data2;
+        }
+    }
+
+    SuggestedPartner.fields = {
+        id: attr(),
+        name: attr(),
+        email: attr(),
+        checked: attr({ default: true }),
+        partner: many2one('mail.partner'),
+        reason: attr(),
```

Comment:
Missing document on all fields.
Also explain why we would need `name`/`email` when there's partner.
I suppose it's when suggested recipient doesn't have a partner yet.
Some of these fields should be computed in case of relation with partner, so that they act like related.

Reply:
> I suppose it's when suggested recipient doesn't have a partner yet.

This is true. It's the while purpose of this model

> Some of these fields should be computed in case of relation with partner, so that they act like related.

All of them are filled by the backend already. And the template doesn't show them if there is a partner. This part doesn't look usefull to me.

Reply:
If the name of partner changes, `partner.name` is updated, and we should expect that name here is the same (for consistency).

It's best not to spread the same info at different places without any sync, because we don't want inconsistencies.
And maybe 99.9% of the time right now no partner has a name or email change during this time, so could motivate it works at least 99.9% of the time right now (and maybe 0.1% doesn't make crash... but that's a fortunate coincidence with current code). I don't like to rely on "_practically_ works _almost_ all the time" or "it works with _current_ flow": 

Good modeling should be declarative in a way that it works with any (ofc. functionally meaningful) flow.
_(infinite amount, including new imps or alternative way of implementing something that has same functionality)_ 

It's not about not trusting the server, but it's not really possible to design good/interesting client application without having some (smart) state management system on client side, such as making data as consistent as possible.